src/protocol.rs

```diff
@@ -450,6 +470,66 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    #[test]
+    fn participant_config_has_more_than_one_participant() {
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let _p = ParticipantConfig::random_quorum(0, &mut rng);
+        });
+
+        match result {
+            Ok(_) => {
+                println!("No panic occurred.");
+            }
+            Err(panic) => {
+                if let Some(s) = panic.downcast_ref::<String>() {
+                    println!("Panic occurred: {s}");
+                    assert_eq!(s, "Received a message with too few parties (Participant Config should have at least 2 parties)");
+                }
+            }
+        }
+
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let _p = ParticipantConfig::random_quorum(1, &mut rng);
+        });
+
+        match result {
+            Ok(_) => {
+                println!("No panic occurred.");
+            }
+            Err(panic) => {
+                if let Some(s) = panic.downcast_ref::<String>() {
+                    println!("Panic occurred: {s}");
+                    assert_eq!(s, "Received a message with too few parties (Participant Config should have at least 2 parties)");
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn participant_does_not_have_empty_set_of_other_participants() {
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let QUORUM_SIZE = 1;
+            let config = ParticipantConfig::random(QUORUM_SIZE, &mut rng);
+            let auxinfo_sid = Identifier::random(&mut rng);
+            let _p = Participant::<AuxInfoParticipant>::from_config(&config, auxinfo_sid, ());
+        });
+
+        match result {
+            Ok(_) => {
+                println!("No panic occurred.");
+            }
```

Comment:
todo: same issue as the other test here: this allows an unacceptable outcome (`from_config` returned `Ok` with an input that was supposed to fail) happen without the test actually failing.